centreon/www/include/configuration/configObject/hostgroup/DB-Func.php

```diff
@@ -152,30 +152,109 @@ function removeRelationLastHostgroupDependency(int $hgId): void
     }
 }
 
-function deleteHostGroupInDB($hostGroups = array())
+function deleteHostGroupInDB(bool $isCloudPlatform, array $hostGroups = [])
 {
     global $pearDB, $centreon;
 
     foreach (array_keys($hostGroups) as $hostgroupId) {
         $previousPollerIds = getPollersForConfigChangeFlagFromHostgroupId((int) $hostgroupId);
 
         removeRelationLastHostgroupDependency((int) $hostgroupId);
-        $rq = "SELECT @nbr := (SELECT COUNT( * ) FROM host_service_relation WHERE service_service_id = " .
-            "hsr.service_service_id GROUP BY service_service_id ) AS nbr, hsr.service_service_id FROM " .
-            "host_service_relation hsr WHERE hsr.hostgroup_hg_id = '" . $hostgroupId . "'";
-        $dbResult = $pearDB->query($rq);
+        $rq = <<<'SQL'
+            SELECT @nbr := (
+                SELECT COUNT( * )
+                FROM host_service_relation
+                WHERE service_service_id = hsr.service_service_id
+                GROUP BY service_service_id
+            ) AS nbr,
+                hsr.service_service_id
+            FROM host_service_relation hsr
+            WHERE hsr.hostgroup_hg_id = :hostgroup_id
+            SQL;
+        $stmt= $pearDB->prepare($rq);
+        $stmt->bindValue(":hostgroup_id", (int) $hostgroupId, \PDO::PARAM_INT);
+        $stmt->execute();
 
         $statement = $pearDB->prepare("DELETE FROM service WHERE service_id = :service_id");
-        while ($row = $dbResult->fetch()) {
+        while ($row = $stmt->fetch()) {
             if ($row["nbr"] == 1) {
                 $statement->bindValue(':service_id', (int) $row["service_service_id"], \PDO::PARAM_INT);
                 $statement->execute();
             }
         }
-        $dbResult3 = $pearDB->query("SELECT hg_name FROM `hostgroup` WHERE `hg_id` = '" . $hostgroupId . "' LIMIT 1");
-        $row = $dbResult3->fetch();
 
-        $pearDB->query("DELETE FROM hostgroup WHERE hg_id = '" . $hostgroupId . "'");
+        $statement = $pearDB->prepare(
+            <<<'SQL'
+                SELECT hg_name
+                FROM hostgroup
+                WHERE hg_id = :hostgroup_id
+                LIMIT 1
+                SQL
+        );
+        $statement->bindValue(':hostgroup_id', (int) $hostgroupId, \PDO::PARAM_INT);
+        $statement->execute();
+        $row = $statement->fetch();
+
+        if ($isCloudPlatform) {
+            if ($pearDB->beginTransaction()) {
```

Comment:
as you don't have any else, you could do if ($isCloudPlatform && $pearDB->beginTransaction()) { . . . }